docs/features/scripting.rst

```diff
@@ -84,6 +84,28 @@ command. Here's a simple example that uses the arg_printer_ script::
 system paths, and as shown above it has the ability to pass command-line
 arguments to the scripts invoked.
 
+Developing a CMD2 API 
+---------------------
+
+If you as an app designer have not explicitly disabled the run_pyscript command it must be assumed
+that your application is structured for use in higher level python scripting. The following sections
+are meant as guidelines and highlight possible pitfalls with both production and consumption
+of API functionality. For clarity when speaking of "scripter" we are referring to those writing
+scripts to be run by pyscript and "designer" as th CMD2 application author. 
+
+Basics
+~~~~~~
+
+When implementing a command, setting ``self.last_result`` allows for application-specific
+data to be returned to a python script from the command. This can allow python scripts to
+make decisions based on the result of previous application commands.
+
+The application command (default: ``app``) returns a ``cmd2.CommandResult`` for each command.
+The ``cmd2.CommandResult`` object provides the captured output to ``stdout`` and ``stderr``
+while a command is executing. Additionally, it provides the value that command stored in
+``self.last_result``.
```

Comment:
Specify that the `CommandResult` object stores `self.last_result` in its `data` member.
Also mention `CommandResult.stop` which is the return value of `onecmd_plus_hooks()` after it runs the given command line. It says whether the command loop is intended to stop. (e.g. quit returns `True`)

Reply:
CommandREsult.stop seems extra? how is that different than stderr or CommandResult.__bool__? I know that they are different but I'm not groking how I as a scriptor would use one over another?

Reply:
Without any work on the part of the designer, a scripter can take advantage of piecing together workflows
using simple ``app`` calls. The result of a run_pyscript app call yields a ``CommandResult`` object exposing 
four members: ``Stdout``, ``Stderr``, ``Stop``, and ``Data``. 

``Stdout`` and ``Stderr`` are fairly straightforward representations of normal data streams and accurately reflect
what is seen by the user during normal cmd2 interaction. ``Stop`` contains information about how the invoked 
command has ended its lifecycle. Lastly ``Data`` contains any information the designer sets via ``self.last_result`` 
or ``self._cmd.last_result`` if called from inside a CommandSet.